proxy/proxy.go

```diff
@@ -235,13 +240,19 @@ func newCattleProxies(config *Config) (*proxyProtocolConverter, *cattleWSProxy)
 }
 
 type proxyProtocolConverter struct {
-	reverseProxy *httputil.ReverseProxy
-	httpsPorts   map[int]bool
+	reverseProxy        *httputil.ReverseProxy
+	httpsPorts          map[int]bool
+	apiFilterProxy      *apifilterproxy.APIFiltersHandler
+	apiFilterProxyReady bool
 }
 
 func (h *proxyProtocolConverter) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	proxyprotocol.AddHeaders(req, h.httpsPorts)
-	h.reverseProxy.ServeHTTP(rw, req)
+	if h.apiFilterProxyReady {
```

Comment:
On that note, I don't think you should have this switch logic here. the cattleProxy should be a property on your proxy and in your proxy, if no path is matched, you should called the cattleProxy